Source/WebCore/inspector/InspectorStyleSheet.cpp

```diff
@@ -1537,22 +1506,62 @@ ExceptionOr<void> InspectorStyleSheet::setRuleStyleText(const InspectorCSSId& id
         ASSERT(childStart <= childEnd);
         RELEASE_ASSERT_WITH_SECURITY_IMPLICATION(childEnd <= styleSheetText.length());
 
-        replacementBodyText.append('\n', indentation);
+        canonicalRuleText.append('\n', indentation);
 
         // Non-style rules don't include the `@rule` prelude in their header range.
         if (childRuleSourceData->type != StyleRuleType::Style)
-            replacementBodyText.append(atRuleIdentifierForType(childRuleSourceData->type));
+            canonicalRuleText.append(atRuleIdentifierForType(childRuleSourceData->type));
 
-        replacementBodyText.appendSubstring(styleSheetText, childStart, childEnd - childStart);
-        replacementBodyText.append("}\n"_s);
+        canonicalRuleText.appendSubstring(styleSheetText, childStart, childEnd - childStart);
+        canonicalRuleText.append("}\n"_s);
     }
 
-    auto closingIndentationLineStart = newText.reverseFind('\n');
+    auto closingIndentationLineStart = ruleStyleDeclarationText.reverseFind('\n');
     if (closingIndentationLineStart != notFound)
-        replacementBodyText.appendSubstring(newText, closingIndentationLineStart);
+        canonicalRuleText.appendSubstring(ruleStyleDeclarationText, closingIndentationLineStart);
+
+    return canonicalRuleText;
+}
+
+ExceptionOr<void> InspectorStyleSheet::setRuleStyleText(const InspectorCSSId& id, const String& newStyleDeclarationText, String* oldStyleDeclarationText, const String* newRuleText, String* oldRuleText)
```

Comment:
Nit: newRuleText should be a const String&
Nit: name the String* arguments to be more obviously out-params. i.e., `returnedOldRuleText`, `out_oldRuleText`, or similar.

Reply:
> Nit: newRuleText should be a const String&

It could be null for undoing, where we want to avoid computing the canonical rule text. I can add a short comment to signal that.